### ISSUE TITLE
It's difficult to write tests

### ISSUE THREAD

**Issue by sitarane** [markets/invisible_captcha#126]
I think we need some improvements on the doc on the topic of tests. I'm happy to contribute it, but i have a few questions first.

When I write a request test for a POST request against a controller protected by invisible_captcha, there are two issue:
1. If the field name is random, I can't gess what it's going to be.
2. The spinner param is expected, but it's unclear what its value should be.

My solution was to put this in my initializer:

```ruby
config.honeypots = ['botfield'] if Rails.env.test?
config.spinner_enabled = !Rails.env.test?
```

That way I know that the honeypot field in tests is always "botfield".

But there might be a better way to do this.

Anyone already figured this out?

**Comment by mishaelajay** [comment]
I am facing the same issue.

**Comment by markets** [comment]
Hey @sitarane sorry for this late response 🙏🏼, GH notifications system doesn't help at all ...

About this topic, we already have a section in the README explaining some techniques: https://github.com/markets/invisible_captcha?tab=readme-ov-file#testing-your-controllers. But yeah, maybe we can expand this section by adding more tricks. 

Regarding your doubts/changes for testing:
1) **random honeypots**: I think your approach is just enough
2) **spinner**: I think it's also fine to disable it for testing, but in case you want to have it enabled, this validation relies on the following check:

https://github.com/markets/invisible_captcha/blob/eb72f0db2222e5f754eee1ed816ff0050640ff5d/lib/invisible_captcha/controller_ext.rb#L78

So basically, a value in your `session` object and a value in a hidden input should match. Internally, it uses a configurable seed to encode those values. Please see an example on current specs:

https://github.com/markets/invisible_captcha/blob/eb72f0db2222e5f754eee1ed816ff0050640ff5d/spec/controllers_spec.rb#L226-L247

Feel free to send documentation updates regarding those couple of points.

**Comment by markets** [comment]
@sitarane @mishaelajay I finally pushed this: https://github.com/markets/invisible_captcha/commit/a6fce921bd675923520df5d6434ca62f2f2b9e51. Hope it helps! Feel free to send any feedback.
